### PR TITLE
Tsc 442 front end for admin

### PR DESCRIPTION
**Pull request by paologum**
front end for adding and deleting and getting users
also changed one migration file to have userId be unique
changed `fastifyCompress` to have threshold due to bug where it doesn't send info below certain threshold
refactored some components

# light mode and add user

https://github.com/user-attachments/assets/a7bf52cd-768b-44a7-aeea-e31a935ea83b

# light mode and delete user and grid capabilities

https://github.com/user-attachments/assets/04f908d9-fc4b-4b22-8013-f9fd852d0a35

# dark mode

image

### CODE REVIEW

File: app/src/admin/Admin.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      tabName: "User Config",
+      component: <AdminUserConfig />
+    },
+    {
+      tabName: "Datapack Config",
+      component: <AdminDatapackConfig />
+    }
+  ];
+  return (
+    <Box ml="50px" mt="50px" display="flex" flexDirection="column">
```
Comment (jay-jhk-lee) [inline app/src/admin/Admin.tsx:53]: Is there a reason you don't have this in a class

File: app/src/admin/AdminUserConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        defaultColDef={defaultCol}
+        ref={gridRef}
+        isRowSelectable={(node) => node.data.email !== state.user.email}
+        rowMultiSelectWithClick
+        rowSelection="multiple"
+        columnDefs={colDefs}
+        rowData={state.admin.displayedUsers}
+        rowDragManaged={true}
+      />
+      <Box display="flex" flexDirection="row" gap="10px" margin="auto" mt="10px" mb="10px">
```
Comment (jay-jhk-lee) [inline app/src/admin/AdminUserConfig.tsx:109]: same

**Review by jay-jhk-lee — COMMENTED** [review]
Looks Good, some styling questions. The ag grid looks sleek

**Review by jay-jhk-lee — APPROVED** [review]

File: server/src/admin-routes.ts [hunk 3]
```diff
@@ -28,10 +29,21 @@ export const getUsers = async function getUsers(_request: FastifyRequest, reply:
     const displayedUsers = users.map((user) => {
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       const { hashedPassword, ...displayedUser } = user;
```
Comment (asivath) [inline server/src/admin-routes.ts:31]: don't need this eslint thing

File: app/src/state/actions/admin-actions.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      pushError(ErrorCodes.RECAPTCHA_FAILED);
+      return;
+    }
+  } catch (error) {
+    pushError(ErrorCodes.RECAPTCHA_FAILED);
+    return;
+  }
+  try {
+    const response = await fetcher("/admin/users", {
+      method: "GET",
```
Comment (asivath) [inline app/src/state/actions/admin-actions.ts:23]: get or post

File: server/src/admin-routes.ts (hunk 3 above)
Comment (paologum) [reply server/src/admin-routes.ts:31]: wait i do need this because i don't use `hashedPassword`

**Pull request merged**